fabric-networking-handshake-v1/src/main/java/net/fabricmc/fabric/impl/networking/handshake/LanguageInjection.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.networking.handshake;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import net.minecraft.util.JsonHelper;
+
+import net.fabricmc.loader.api.FabricLoader;
+import net.fabricmc.loader.api.ModContainer;
+
+// todo move to resources maybe
+public final class LanguageInjection {
+	private static final Logger LOGGER = LogManager.getLogger();
+
+	private LanguageInjection() {
+	}
+
+	public static void injectFabricDefaultTranslations(Map<String, String> translations, Pattern argPattern) {
+		for (ModContainer mod : FabricLoader.getInstance().getAllMods()) {
+			Path langFile = mod.getPath("assets").resolve(mod.getMetadata().getId()).resolve("lang").resolve("en_us.json");
```

Comment:
Injecting a few message is complicated since you need to declare what language files you need to inject.

The reason I inject only the lang file under the mod namespace is vanilla do the same (vanilla has lang files for both `minecraft` and `realms` namespace, but only lang file in `minecraft` namespace is injected)

An alternative is to move this feature to resource loader v0 and make the handshake mod depend on resource loader (in mod json, exposed as implementation configuration). @jamieswhiteshirt thoughts on this resource loader migration approach?

Reply:
Handshaking has been postponed.